packages/edit-navigation/src/components/menus-editor/index.js

```diff
@@ -56,6 +101,8 @@ export default function MenusEditor( { blockEditorSettings } ) {
 						setStateMenus( newStateMenus );
 						if ( newStateMenus.length ) {
 							setMenuId( newStateMenus[ 0 ].id );
+						} else {
+							setMenuId();
```

Comment:
This fixes a separate bug that when you delete the last menu, the menuId still refers to it.